buffer.c

```diff
@@ -199,18 +114,18 @@ PHP_METHOD(ArrayBuffer, __construct)
 	long length;
 	zend_error_handling error_handling;
 
-	zend_replace_error_handling(EH_THROW, NULL, &error_handling TSRMLS_CC);
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &length) == FAILURE) {
-		zend_restore_error_handling(&error_handling TSRMLS_CC);
+	zend_replace_error_handling(EH_THROW, NULL, &error_handling);
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &length) == FAILURE) {
```

Comment:
Use zend_parse_parameters_throw instead of replacing error handling.

Reply:
I changed the other cases where it stopped error handling but there are other `zend_parse_parameters` that just return on failure. (I looked at the header file and `zend_parse_parameters` and `zend_parse_parameters_throw` are currently equivalent but maybe you want to signal clearly all errors in parameter parsing throw exceptions.)

Reply:
zend_parse_parameters_throw is the same as zend_parse_parameters in PHP 8, because throw is not the only option. Previously zend_parse_parameters_throw was required to throw on invalid arguments.